bibtex-actions.el

```diff
@@ -417,10 +421,67 @@ TEMPLATE."
 ;; something equivalent.
 (defun bibtex-actions-get-key-org-cite ()
   "Return key at point for org-cite citation-reference."
-  (when (eq major-mode 'org-mode)
-    (let ((elt (org-element-context)))
-      (if (eq (car elt) 'citation-reference)
-          (org-element-property :key elt)))))
+  (when-let (((eq major-mode 'org-mode))
+             (elt (org-element-context)))
+    (pcase (org-element-type elt)
+      ('citation-reference
+       (org-element-property :key elt))
+      ('citation
+       (org-cite-get-references elt t)))))
+
+;;; Embark
+
+(defun bibtex-actions-citation-key-at-point ()
+  "Return citation keys at point as string for `embark'."
+  (when-let ((key (or (bibtex-actions-get-key-org-cite)
+                      (bibtex-completion-key-at-point))))
+    (cons 'citation-key (if (listp key) (string-join key ";") key))))
+
+(with-eval-after-load 'embark
+  (add-to-list 'embark-target-finders 'bibtex-actions-citation-key-at-point)
+  (add-to-list 'embark-keymap-alist '(citation-key . bibtex-actions-embark-map))
+  (add-to-list 'embark-keymap-alist '(bibtex . bibtex-actions-map)))
+
+(defvar bibtex-actions-embark-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "t") 'bibtex-actions-embark-add-pdf-attachment)
+    (define-key map (kbd "a") 'bibtex-actions-embark-add-pdf-to-library)
+    (define-key map (kbd "o") 'bibtex-actions-embark-open)
+    (define-key map (kbd "e") 'bibtex-actions-embark-open-entry)
+    (define-key map (kbd "l") 'bibtex-actions-embark-open-link)
+    (define-key map (kbd "n") 'bibtex-actions-embark-open-notes)
+    (define-key map (kbd "p") 'bibtex-actions-embark-open-pdf)
+    (define-key map (kbd "RET") 'bibtex-actions-embark-open)
+    map)
+  "Keymap for Embark citation-key actions.")
+
+(defun bibtex-actions-embark-open (keys)
```

Comment:
Next big question: why do we need these functions? 

Can you not just have a little function to split if a string, and leave alone otherwise, so that you can remove them entirely?

Aside from great maintenance burden and less flexibility, it results in really long names that are a problem in which-key, at least.

![Screenshot from 2021-06-13 11-09-34](https://user-images.githubusercontent.com/1134/121812973-0dff4480-cc38-11eb-9597-fddbc3096085.png)

BTW, if any has a great shorter and more general (non-bibtex-specific) name for this package ... :-)

Reply:
I agree that `bibtex-actions-embark-*` functions should be removed finally.

> ... and then something like:

It doesn't seem to work as expected. The tricky part is the entries are always chosen via `completing-read` when `bibtex-actions-add-pdf-to-library` is interactively called. `embark` also interactively calls `bibtex-actions-add-pdf-to-library`, which means the `keys` found at point are not used by `bibtex-actions-add-pdf-to-library`. I need to find a proper way to handle this.

Reply:
I hadn't thought about the "add" commands. Could always defer those to a later PR.

Reply:
The problem is the same for `bibtex-actions-open-*` actually.